lib/rspec/rails/feature_check.rb

```diff
@@ -43,8 +43,11 @@ def has_action_mailbox?
         defined?(::ActionMailbox)
       end
 
-      def ruby_3_1?
-        RUBY_VERSION >= "3.1"
+      # TODO: add a self-explanatory method name. See https://github.com/rspec/rspec-rails/pull/2552#issuecomment-1009508693 for hints
+      def rails_6_1_and_ruby_3_1?
+        return false if RUBY_VERSION < "3.1"
+
+        ::Rails::VERSION::STRING >= "6.1" && ::Rails::VERSION::STRING < "7"
```

Comment:
I haven't seen this in practice, but it's tempting to write this as:
```ruby
"6.1" <= ::Rails::VERSION::STRING < "7"
```

Reply:
I'd probably avoid that.

Simpler is usually better than more concise.  I'd want a developer to be able to look at it and immediately know what it means.  And with the proposed change I don't think that'd be true.